index.html

```diff
@@ -876,41 +876,95 @@ <h3>Directory Service API</h3>
                     incremental transfer of the payloads.
                 </p>
 
+                <p>
+                    The directory APIs include mandatory, recommended, and optional
+                    features. 
+                    Directories that don't offer recommended or optional features
+                    inform the clients about the absence of those features
+                    according to the following rules:
+                    <ul>
+                        <li>
+                            If the missing feature is to customize existing functionality of an
+                            API, the server will respond with 501 (Not Implemented)
+                            HTTP status. For example, if a client requests a list of
+                            items with a particular sorting order, the server either
+                            respects that request and returns the list with the requested
+                            order, or rejects the request and does not return the list.
+                            The server will return the list with the default order
+                            if and only if custom sorting is not requested.
+                        </li>
+                        <li>
+                            If the missing feature is the whole functionality at an API endpoint,
+                            the server may respond with 404 (Not Found).
+                            Alternatively, a server may respond with 501 (Not Implemented)
+                            to signify that it has placeholders in place for the given
+                            endpoint and would provide this functionality in future.
```

Comment:
I wasn't sure if this should be an assertion. It won't help the clients anyway because it is optional for servers. Turning it into an assertion also means that we need to find two implementations who don't implement the feature and use those status codes!

Reply:
I am not completely convinced about returning `501` either. Programmatically speaking having `404` or `501` result in the same output -> I cannot use this endpoint. About user, perspective knowing that the feature Might be implemented in the future feels like a temporary workaround. I would prefer chooseing just one of the two error codes.

Reply:
Removed 501 and converted 404 to an assertion, but recommended instead of optional.

Reply:
OK.  I agree with only having one error code.  I am personally unsure why we don't make this mandatory but having it as a recommended is ok (probably best to not force an implementation to do anything at all if there is an error).

Reply:
The "may" should probably be MAY though and marked as an assertion with a span and id, however.  Are all errors written like this?  Do we have a general statement/assertion somewhere about errors that makes individual assertions unnecessary?

Reply:
@mmccool I think you are commenting on the outdated diff.

Regarding mandatory or recommended; as you said, forcing this means server have to implement it specifically that way. For example, a server may need to implement all those HTTP handlers and return 404.